src/appProvider.js

```diff
@@ -22,13 +22,16 @@ class AppProvider extends React.Component {
     super(props);
```

Comment:
we should just remove the constructor all together and just do
```
state = {
  locale: localStorage.getItem('localindex') || 0,
}
```

Reply:
then we can remove the `defaultidx` variable that's not necessary since it's only used once...